### PR TITLE
A task to create a list of BC ranges

### PR DESCRIPTION
**Pull request by pbuehler**
selectBCRange.cxx contains a task which takes the event filter decisions as input and creates a list BCs to be saved.

### CODE REVIEW

**Comment by pbuehler** [comment]
This is a prototype and NOT to be merged yet.

**Comment by mpuccio** [comment]
> This is a prototype and NOT to be merged yet.

I converted the PR to draft such that we do not merge it by mistake. Whenever you think it is ready we can convert it back

**Review by shahor02 — COMMENTED** [review]
Hi @pbuehler 
I've run 
```
copts="--configuration json://BCRangesConfig.json"
o2-analysis-timestamp $copts | o2-analysis-event-selection $copts | o2-analysis-selected-bc-range-task $copts
```
on AO2D from recent MC production, and on this level it works.
But when I attach to it the irframes writer (needs https://github.com/AliceO2Group/AliceO2/pull/8483):
```
o2-analysis-timestamp $copts | o2-analysis-event-selection $copts | o2-analysis-selected-bc-range-task $copts | o2-irframe-writer-workflow --input-spec "x:PPF/BCR/0"
```
it crashes with 
```
[881689:bc-selection-task]: [20:35:04][ERROR] GetChannel(): 'from_bc-selection-task_to_bc-ranges-selector-task[0]' does not exist.
[881690:event-selection-task]: [20:35:04][INFO] Processing timeslice:0, tfCounter:0, firstTFOrbit:0, action:0
[881690:event-selection-task]: [20:35:04][INFO] Creating 0.338816MB for table AOD/EVSEL.
```
Looks like the channel for the forwarding of `AOD/BCSEL/0` is not created in this case:
Screenshot from 2022-03-31 20-36-40

This looks like a DPL problem, I've encountered similar situation once and it was not yet resolved. I think we need @ktf 's intervention here.

File: EventFiltering/selectBCRange.cxx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    InteractionRecord IR1, IR2;
+    LOGF(debug, "Merged and extended sorted ranges");
+    for (auto limit : cbcrs.list()) {
+      LOGF(debug, "  %i - %i", limit.first, limit.second);
+      IR1.setFromLong(limit.first);
+      IR2.setFromLong(limit.second);
+      res.emplace_back(IR1, IR2);
+    }
+    // make res an output
+    pc.outputs().snapshot({"PPF", "BCR", 0, Lifetime::Timeframe}, res);
```
[the remark this answers is not in the document]
Comment (shahor02) [reply EventFiltering/selectBCRange.cxx:90]: For @ktf : uploading json workflows generated by:
```
o2-analysis-timestamp $copts | o2-analysis-event-selection $copts | o2-analysis-selected-bc-range-task $copts  --dump > wfOK.json
```
and 
```
o2-analysis-timestamp $copts | o2-analysis-event-selection $copts | o2-analysis-selected-bc-range-task $copts | o2-irframe-writer-workflow --input-spec "x:PPF/BCR/0" --dump > wfFail.json
```
[wfOK.json.gz](https://github.com/AliceO2Group/O2Physics/files/8397381/wfOK.json.gz)
[wfFail.json.gz](https://github.com/AliceO2Group/O2Physics/files/8397382/wfFail.json.gz)

[the remark this answers is not in the document]
Comment (pbuehler) [reply EventFiltering/selectBCRange.cxx]: @mpuccio @shahor02
Hi Max,
This PR has been discussed and tested with Ruben and is ready to be merged. 
Currently it is assumed, that all collisions are accepted. When the filter table by Stefano is ready I will include the filter decisions.

**Comment by pbuehler** [comment]
@mpuccio @shahor02
Hi Max,
This PR has been discussed and tested with Ruben and is ready to be merged. 
Currently it is assumed, that all collisions are accepted. When the filter table by Stefano is ready I will include the filter decisions.

**Review by mpuccio — DISMISSED** [review]

**Comment by mpuccio** [comment]
@TimoWilken clang-format and the copyright header check are stuck since 10 minutes, are there any issues?

**Comment by mpuccio** [comment]
@pbuehler can you try to force push here to re-trigger the CI?

**Comment by TimoWilken** [comment]
@mpuccio @pbuehler, sorry, the formatting workflow changed significantly, so could you rebase your changes on top of the current `master` branch and force-push?

**Comment by pbuehler** [comment]
+1

**Review by mpuccio — APPROVED** [review]

**Pull request merged**